server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -350,47 +351,27 @@ public ProjectMetadata addComponentTemplate(
             }
         }
 
-        final Template finalTemplate = Template.builder(template.template()).settings(finalSettings).mappings(wrappedMappings).build();
-        final long now = instantSource.instant().toEpochMilli();
-        final ComponentTemplate finalComponentTemplate;
-        if (existing == null) {
-            finalComponentTemplate = new ComponentTemplate(
-                finalTemplate,
-                template.version(),
-                template.metadata(),
-                template.deprecated(),
-                now,
-                now
-            );
-        } else {
-            final ComponentTemplate templateToCompareToExisting = new ComponentTemplate(
-                finalTemplate,
-                template.version(),
-                template.metadata(),
-                template.deprecated(),
-                existing.createdDateMillis().orElse(null),
-                existing.modifiedDateMillis().orElse(null)
-            );
-            if (templateToCompareToExisting.equals(existing)) {
-                return project;
-            }
-            finalComponentTemplate = new ComponentTemplate(
-                finalTemplate,
-                template.version(),
-                template.metadata(),
-                template.deprecated(),
-                existing.createdDateMillis().orElse(null),
-                now
-            );
-        }
+        final Long now = instantSource.instant().toEpochMilli();
+        final Long createdDateMillis = existingTemplate == null ? now : existingTemplate.createdDateMillis().orElse(null);
+        final ComponentTemplate finalComponentTemplate = new ComponentTemplate(
+            template.template(),
+            template.version(),
+            template.metadata(),
+            template.deprecated(),
+            createdDateMillis,
+            now
+        );
 
-        validateTemplate(finalSettings, wrappedMappings, indicesService);
+        // These two validation checks are only scoped to the component template itself (and don't depend on any other entities in the
+        // cluster state) and could thus be done in the transport action. However, since we're parsing mappings here, we shouldn't be doing
+        // it directly on the transport thread. Instead, we should fork to a different threadpool (management/generic).
+        validateTemplate(finalComponentTemplate.template().settings(), finalComponentTemplate.template().mappings(), indicesService);
         validate(name, finalComponentTemplate.template(), List.of(), null);
```

Comment:
I experimented with moving these two validations to the transport action, but since the Distributed Coordination team advised running them on the management or generic threadpool, I decided to leave it as future work. In terms of overall processing, we wouldn't save much. The only/main win would be that we avoid adding a cluster state update task in case the validation fails.

Also worth noting is that we potentially do more (heavier) mappings validation in `validateIndexTemplateV2` below. Although #135457 helps reduce that in a lot of cases.